bms-monitor.js

```diff
@@ -1,18 +1,38 @@
 const {expect} = require('chai');
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
+isTemperatureOutOfRange = function(temperature) {    
+    let isOutOfRange = false;
+    if (temperature > 45 || temperature < 0) {
         console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
+        isOutOfRange = true;
+    }
+    return isOutOfRange;
+}
+
+isSocOutOfRange = function(soc) {
+    let isOutOfRange = false;
+    if (soc > 80 || soc < 20) {

```

Comment:
Instead of repeating the `<` and `>` checks in each function, isolate the range-checking in a separate function (like `isOutOfRange`). Pass all required parameters to it. Then any enhancements to the range-computation could be done in one place.

Reply:
Sure !! Updated the code accordingly. Thanks for your feedback